editor/src/core/shared/element-template.ts

```diff
@@ -763,6 +763,63 @@ export function setJSXAttributesAttribute(
   return result
 }
 
+export function attributeReferencesElsewhere(attribute: JSXAttribute): boolean {
+  switch (attribute.type) {
+    case 'ATTRIBUTE_VALUE':
+      return false
+    case 'ATTRIBUTE_OTHER_JAVASCRIPT':
+      return (
+        attribute.definedElsewhere.length > 0 ||
+        Object.values(attribute.elementsWithin).some((element) => {
+          return elementReferencesElsewhere(element)
+        })
```

Comment:
I'm pretty much certain we can skip this inner check, because IIRC the inner `definedElsewhere` will either be coming from this JS block or will be coming from this block's `definedElsewhere`

Reply:
That sounds like an axiom which should be covered by a unit test.

Reply:
Unit test confirms this axiom to be true: https://github.com/concrete-utopia/utopia/pull/2416